geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala

```diff
@@ -84,7 +84,9 @@ case class QueryPlanner(sft: SimpleFeatureType,
     val dedupedFeatures = if (deduplicate) new DeDuplicatingIterator(features) else features
 
     val sortedFeatures = if (query.getSortBy != null && query.getSortBy.length > 0) {
-      new LazySortedIterator(dedupedFeatures, query.getSortBy)
+      // TODO: is there a better place to compute the return SFT and propagate it through the query execution?
```

Comment:
I've integrated this PR into the bin aggregation PR - i've modified the query hints there, so it's easy to stick the return sft into it.